docs/customization.md

```diff
@@ -169,3 +199,40 @@ you genuinely intend to remove the Functions Framework console logging.
 
 Logging for tests is provided automatically in the `FunctionTestServer`; see [the testing
 documentation](testing.md) for more details.
+
+## Startup class mismatches
+
+If you haven't received an error referring you to this section of
+the documentation, you *may* want to skip this section.
+
+It is possible (but very rare) for startup classes to affect which
```

Comment:
Ahhh, there's a use case that could be until now, and now can't (a little far fetch though)

What if StartupA, say, depending on configuration, were to add another StartupB on it's own ConfigureServices? StartupB just overwrites the Configure method to add a few things to the pipeline later. That's sort of valid (and StartupB would have actually been executed properly) and sort of makes sense, but would fail now where it didn't before. Maybe we can allow for startup additions with the warning that, the ConfigureConfiguration override won't execute because it's already happened?

I'm actually torn between what should be invalid, a change in startups or a change in function type? Maybe later this can be configurable, like allowFunctionSwap or allowStartupAddition?

Reply:
Adding a startup is really dodgy, because it would only be used for middleware, not config or dependencies. I don't think that's something we want to support, and any previous support is accidental.

We can make this sort of check optional later on, maybe... but I think it's massively corner-case territory.